### PR TITLE
Dashboard version 2  - Back port downstream features, provide ability enable features.

### PR DESCRIPTION
**Pull request by dlabaj**
Description:
Back port Redhat data services dashboard code to ODH.  

Issues Addressed: #194 #189 and #43.

The PR back ports Redhat data services dashboard code to add the following features:

1.  BYON image stream version that is currently shipping with RHODS.
2.  Ability to track users using segment.io (By default this is turned off)
3.  Ability to add badges to Enable and explore cards.
4.  Bug fixes from the downstream repo.
5.  Ability to disable Cluster settings
6.  Add the ability to enable and disable features using odh-dashboard-config config map.

README has been updated to describe how to enable and disable features and their default values.

This PR is dependent on #222

A video demoing the new dashboard can be found [here](https://bluejeans.com/s/fL8QEPSDPX7)

### CODE REVIEW

**Comment by shgriffi** [comment]
@dlabaj this is a pretty big PR.  Do we need any documentation, tutorial or demo to go along with it?  Also, is there a summary of what's new or some screenshots showing the difference?

**Comment by dlabaj** [comment]
> @dlabaj this is a pretty big PR :-) . Do we need any documentation, tutorial or demo to go along with it? Also, is there a summary of what's new or some screenshots showing the difference?

I'll add some additional documentation around enabling and disabling features using the config map.  Will also add a few screen shots.  I added the list of features that where added from RHODS above.  We had a demo today in the ODH community meeting.  We can definitely demo this again.

**Comment by cfchase** [comment]
Perhaps we should get together a kfdef for testing?  It seems like that's a good requirement for all major PRs.  Opinions @LaVLaS ?

**Comment by LaVLaS** [comment]
> Perhaps we should get together a kfdef for testing? It seems like that's a good requirement for all major PRs. Opinions @LaVLaS ?

That is a good point.  At the very least we should have a deployable manifest in this repo that we can use for odh-dashboard CI

**Comment by shgriffi** [comment]
> 

@dlabaj sweet, I think providing a link to the recording of the community meeting would be great and helpful for me.  Thanks for the reply.

**Comment by dlabaj** [comment]
#224

**Comment by lucferbux** [comment]
> > @dlabaj this is a pretty big PR :-) . Do we need any documentation, tutorial or demo to go along with it? Also, is there a summary of what's new or some screenshots showing the difference?
> 
> @shgriffi I'll add some additional documentation around enabling and disabling features using the config map. Will also add a few screen shots. I added the list of features that where added from RHODS above. We had a demo today in the ODH community meeting. We can definitely demo this again.

@dlabaj That's a great idea, are you gonna create a new folder for documentation? If so I can then add more info that I think could be useful, maybe all the configmaps/secrets used for enabling the admin panel and segment. That's something a few people has asked me before

**Review by maroroman — APPROVED** [review]
Tested on @lucferbux cluster

Everything seems to work as advertised. As mentioned in chat there should be a PR created to address the naming of imagestream types so there aren't conflicts when notebook controler stuff gets in.

**Comment by dlabaj** [comment]
@shgriffi Readme  and added a docs folder.  The community video has been added to this PR, where it demos the changes to UI.   The PR that @LaVLaS did for the kfdef is also included in the description above.   @lucferbux Feel free to add additional docs to the docs folder.

**Comment by lucferbux** [comment]
@dlabaj great, I have them right now, should I add commits to this PR or hold off until is merged and open a new PR?

**Comment by dlabaj** [comment]
@lucferbux you can add commits to this PR.

**Comment by LaVLaS** [comment]
When authentication is not enabled, do we plan to handle the issue that results in the pod log showing `Failed to get groups: HttpError: HTTP request failed`?

**Comment by LaVLaS** [comment]
Right now it's just an error output in the logs. I'll create an issue to track it for follow-up

File: backend/src/routes/api/segment-key/segmentKeyUtils.ts
Comment (cfchase) [inline backend/src/routes/api/segment-key/segmentKeyUtils.ts:14]: delete `console.log`

File: backend/src/routes/api/status/index.ts
Comment (cfchase) [inline backend/src/routes/api/status/index.ts:41]: Don't we need an await or then?  I think getApiResources returns a promise.

**Comment by cfchase** [comment]
You might want to check the data folder for some minor changes in the few files we have in common to absorb those changes.  e.g. "features.json" `gpu-computing` and "applications/jupyterhub.yaml"

File: backend/src/routes/api/status/index.ts
Comment (DaoDaoNoCode) [reply backend/src/routes/api/status/index.ts:37]: @LaVLaS https://github.com/opendatahub-io/odh-dashboard/pull/177/files I think I made this update here. Looks like it's not in this PR.

**Comment by dlabaj** [comment]
> Right now it's just an error output in the logs. I'll create an issue to track it for follow-up

I changed it to only log the information when in log level is set to debug.

File: backend/src/routes/api/status/index.ts
Comment (dlabaj) [reply backend/src/routes/api/status/index.ts:41]: Removed this code it's not doing anything.

Comment (dlabaj) [reply backend/src/routes/api/status/index.ts:41]: Removed this code it's not doing anything.

File: backend/src/routes/api/status/index.ts
Comment (dlabaj) [reply backend/src/routes/api/status/index.ts:37]: Pulled in that commit above @DaoDaoNoCode and @LaVLaS

Comment (dlabaj) [reply backend/src/routes/api/status/index.ts:37]: Pulled in that commit above @DaoDaoNoCode and @LaVLaS

**Review by lucferbux — APPROVED** [review]
/lgtm

File: frontend/src/pages/clusterSettings/ClusterSettings.tsx
Comment (dlabaj) [reply frontend/src/pages/clusterSettings/ClusterSettings.tsx:109]: @LaVLaS @lucferbux Is it fine we address then in another PR?  By default this page isn't available.

Comment (lucferbux) [reply frontend/src/pages/clusterSettings/ClusterSettings.tsx:109]: Yes, im gonna mark this as resolved.

**Pull request merged**
